src/test/scala/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsTest.scala

```diff
@@ -1995,4 +1991,182 @@ class OrganizeImportsTest extends OrganizeImportsBaseTest {
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    class X {
+      val b = Buffer()
+      val lb = ListBuffer()
+    }
+    """
+    }
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldInsertEmptyLineBetweenImportsAndTopModuleDef() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+    object X extends App {
+      Buffer
+      ListBuffer
+    }
+    """ becomes {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    object X extends App {
+      Buffer
+      ListBuffer
+    }
+    """
+    }
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldKeepExistingEmptyLineBetweenImportsAndTopClassDef() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    class X {
+      val b = Buffer()
+      val lb = ListBuffer()
+    }
+    """ isNotModified
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldKeepExistingEmptyLineBetweenImportsAndTopModuleDef() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    object X extends App {
+      Buffer
+      ListBuffer
+    }
+    """ isNotModified
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldReduceExistingEmptyLinesBetweenImportsAndTopClassDefToOne() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+
+    class X {
+      val b = Buffer()
+      val lb = ListBuffer()
+    }
+    """ becomes {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    class X {
+      val b = Buffer()
+      val lb = ListBuffer()
+    }
+    """
+    }
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldReduceExistingEmptyLinesBetweenImportsAndTopModuleDefToOne() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+
+    object X extends App {
+      Buffer
+      ListBuffer
+    }
+    """ becomes {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+
+    object X extends App {
+      Buffer
+      ListBuffer
+    }
+    """
+    }
+  } applyRefactoring organizeWithTypicalParams
+
+  @Test
+  def shouldReduceExistingEmptyLinesWithWhitespacesBetweenImportsAndTopModuleDefToOne() = new FileSet {
+    """
+    /*<-*/
+    package x
+
+    import scala.collection.mutable.Buffer
+    import scala.collection.mutable.ListBuffer
+  
+  
+    object X extends App {
```

Comment:
but on other hand when I made git diff I saw these spaces at the ends of
line

2016-03-03 8:28 GMT+01:00 Wieslaw Popielarski <wpopielarski@gmail.com>:

> now I know why in some test `$` is used :)
>
> 2016-03-02 23:32 GMT+01:00 Simon Schäfer <notifications@github.com>:
>
>> In
>> src/test/scala/scala/tools/refactoring/tests/implementations/imports/OrganizeImportsTest.scala
>> <https://github.com/scala-ide/scala-refactoring/pull/143#discussion_r54804049>
>> :
>>
>> > +    }
>> > +    """
>> > +    }
>> > +  } applyRefactoring organizeWithTypicalParams
>> > +
>> > +  @Test
>> > +  def shouldReduceExistingEmptyLinesWithWhitespacesBetweenImportsAndTopModuleDefToOne() = new FileSet {
>> > +    """
>> > +    /*<-*/
>> > +    package x
>> > +
>> > +    import scala.collection.mutable.Buffer
>> > +    import scala.collection.mutable.ListBuffer
>> > +
>> > +
>> > +    object X extends App {
>>
>> There is no whitespace in this test. You need to add some symbol (like $)
>> at the end, otherwise trailing whitespace is killed by save actions.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/scala-ide/scala-refactoring/pull/143/files#r54804049>
>> .
>>
>
>